hibernate-core/src/main/java/org/hibernate/tuple/entity/EntityMetamodel.java

```diff
@@ -131,8 +133,8 @@
 	public EntityMetamodel(
 			PersistentClass persistentClass,
 			EntityPersister persister,
-			SessionFactoryImplementor sessionFactory) {
-		this.sessionFactory = sessionFactory;
+			PersisterCreationContext creationContext) {
```

Comment:
I'm assuming that changing this SPI is OK in master.

Note to self: when this gets backported, the old version of the method should be deprecated.

Reply:
Again, same as previous.  For this though *maybe* other Hibernate projects use it, but I doubt it